backend/vagrant.pm

```diff
@@ -0,0 +1,190 @@
+# Copyright © 2021 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+package backend::vagrant;
+
+use strict;
+use warnings;
+
+use base 'backend::virt';
+use bmwqemu;
+use testapi qw(get_required_var get_var);
```

Comment:
I suggest to avoid use of the external facing API within backends. I tried to remove all existing uses already in other backends but have not achieved that yet. Please instead of something like `get_var` use `$bmwqemu::$vars{…}`